test/jdk/java/net/httpclient/http2/BurstyRequestsTest.java

```diff
@@ -0,0 +1,259 @@
+ *          sudden rush of HTTP/2 requests
+ * @modules java.net.http/jdk.internal.net.http:+open
+ *          java.net.http/jdk.internal.net.http.common
+ *          java.net.http/jdk.internal.net.http.frame
+ *          java.net.http/jdk.internal.net.http.hpack
+ *          java.net.http/jdk.internal.net.http.quic
+ *          java.net.http/jdk.internal.net.http.quic.packets
+ *          java.net.http/jdk.internal.net.http.quic.frames
+ *          java.net.http/jdk.internal.net.http.quic.streams
+ *          java.net.http/jdk.internal.net.http.http3.streams
+ *          java.net.http/jdk.internal.net.http.http3.frames
+ *          java.net.http/jdk.internal.net.http.http3
+ *          java.net.http/jdk.internal.net.http.qpack
+ *          java.net.http/jdk.internal.net.http.qpack.readers
+ *          java.net.http/jdk.internal.net.http.qpack.writers
+ *          java.logging
+ *          java.base/jdk.internal.net.quic
+ *          java.base/jdk.internal.util
+ *          java.base/sun.net.www.http
+ *          java.base/sun.net.www
+ *          java.base/sun.net
+ *
+ * @library /test/lib /test/jdk/java/net/httpclient/lib
+ * @build jdk.test.lib.net.SimpleSSLContext
+ *        jdk.httpclient.test.lib.http2.Http2TestServer
+ *        jdk.httpclient.test.lib.http2.Http2Handler
+ *        jdk.httpclient.test.lib.http2.Http2TestExchange
+ *        jdk.httpclient.test.lib.http2.Http2TestExchangeSupplier
+ * @run junit ${test.main.class}
+ */
+class BurstyRequestsTest {
+
+    private static final String HANDLER_PATH = "/8326498/";
+
+    private static Field openConnections; // Set<> jdk.internal.net.http.HttpClientImpl#openedConnections
+
+    private static SSLContext sslContext;
+    private static Http2TestServer h2server;
+
+    @BeforeAll
+    static void beforeAll() throws Exception {
+        openConnections = Class.forName("jdk.internal.net.http.HttpClientImpl")
+                .getDeclaredField("openedConnections");
+        openConnections.setAccessible(true);
+
+        sslContext = new SimpleSSLContext().get();
+        h2server = new Http2TestServer(true, sslContext);
+        h2server.setExchangeSupplier(new ExchangeSupplier());
+        h2server.addHandler(new Handler(), HANDLER_PATH);
+        h2server.start();
+        System.err.println("started HTTP/2 server " + h2server.getAddress());
+    }
+
+    @AfterAll
+    static void afterAll() {
+        if (h2server != null) {
+            System.err.println("stopping server " + h2server.getAddress());
+            h2server.stop();
+        }
+    }
+
+    /*
+     * Issues a burst of 100 HTTP/2 requests to the same server (host/port) and expects all of
+     * them to complete normally.
+     * Once these requests have completed, the test then peeks into an internal field of the
+     * HttpClientImpl to verify that the client is holding on to at most 1 connection.
+     */
+    @Test
+    void testOpenConnections() throws Exception {
```

Comment:
Shall we also introduce following tests?

1. Verify secondary request bursts always reuse the pooled connection, and don't change the state of the pool. (Remember that, as reported in the associated ticket, the secondary bursts were causing the orphan connection pile up.)
2. Repeat all tests with a small idle timeout, say, 5s, and ensure that after 5s pool is completely emptied.